Add typelevel.g8 section to docs (#258)
typelevel/sbt-typelevel: docs/index.md

@:@

### Giter8 Template

> 

Reviewer: I think we should move this section higher up :) since the `plugins.sbt` stuff isn't necessary for someone using the g8, nor is the choice between sbt-typelevel and sbt-typelevel-ci-release.

Author: Good call, it's awkward in the middle here, but I wasn't sure if we wanted it front and center.

> The sbt-typelevel project provides a [Giter8 template](http://www.foundweekends.org/giter8/index.html) for quickly starting projects with familiar workflows and best practices.

Reviewer: ```suggestion
We provide a [Giter8 template](http://www.foundweekends.org/giter8/index.html) for quickly starting projects with familiar workflows and best practices.
```

```sh
sbt new typelevel/typelevel.g8
```

> This will help you create a new project with sbt-typelevel, sbt-typelevel-site, and some other boilerplate setup.

Reviewer: sbt-typelevel is Boilerplate Free™ 🤣 

```suggestion
This will guide you through the basic setup to create a new project with **sbt-typelevel** and **sbt-typelevel-site**.
```

> Check out the project repo for more details: [typelevel.g8](https://github.com/typelevel/typelevel.g8)

Reviewer: ```suggestion
Check out the [typelevel.g8](https://github.com/typelevel/typelevel.g8) project for more details.
```
